sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDiskStore.scala

```diff
@@ -82,20 +88,26 @@ class RapidsDiskStore(diskBlockManager: RapidsDiskBlockManager)
       incoming: RapidsBuffer,
       path: File,
       append: Boolean,
-      stream: Cuda.Stream): (Long, Long) = {
+      stream: Cuda.Stream): (Long, Long, Long) = {
     incoming match {
       case fileWritable: RapidsBufferChannelWritable =>
-        withResource(new FileOutputStream(path, append)) { fos =>
-          withResource(fos.getChannel) { outputChannel =>
-            val startOffset = outputChannel.position()
-            val writtenBytes = fileWritable.writeToChannel(outputChannel, stream)
-            if (writtenBytes == 0) {
-              throw new IllegalStateException(
-                s"Buffer ${fileWritable} wrote 0 bytes disk on spill. This is not supported!"
-              )
+        val currentPos = if (!path.exists()) {
+          path.createNewFile()
+          path.length()
+        } else {
+          path.length()
+        }
+        var startOffSet, writtenBytes = 0L
+        withResource(FileChannel.open(Paths.get(path.toURI), StandardOpenOption.APPEND)) { fc =>
```

Comment:
This is blindly appending which means it's now ignoring the `append` parameter.  If `append` is not set, this should write to the beginning of the file, not the end, as file sharing between buffers is not allowed.

Reply:
Oh, my bad. Adding this back. Was mis-reading ```append``` as a file option other than parameter.